dc/core/Markers.cpp

```diff
@@ -39,38 +39,81 @@
 
 #include "Markers.h"
 
-Markers::Markers()
+Markers::Markers( QObject* parent_ )
+    : QAbstractListModel( parent_ )
+{}
+
+QVariant Markers::data( const QModelIndex& index_, int role ) const
+{
+    if( index_.row() < 0 || index_.row() >= rowCount() || !index_.isValid( ))
+        return QVariant();
+
+    if( role == XPOSITION_ROLE )
+    {
+        return QVariant( _markers[index_.row()].second.x( ));
+    }
+    else if( role == YPOSITION_ROLE )
+    {
+        return QVariant( _markers[index_.row()].second.y( ));
+    }
+    return QVariant();
+}
+
+int Markers::rowCount( const QModelIndex& parent_ ) const
 {
+    Q_UNUSED( parent_ );
+    return _markers.size();
 }
 
-const MarkersMap& Markers::getMarkers() const
+QHash<int, QByteArray> Markers::roleNames() const
 {
-    return _markers;
+    QHash<int, QByteArray> roles;
+    roles[ XPOSITION_ROLE ] = "xposition";
+    roles[ YPOSITION_ROLE ] = "yposition";
+    return roles;
 }
 
 void Markers::addMarker( const int id, const QPointF position )
 {
-    if( _markers.count( id ))
+    if( std::find_if( _markers.begin(), _markers.end(), [&id]( Marker& marker )
+        { return marker.first == id; }) != _markers.end( ))
+    {
         return;
+    }
 
-    _markers[id] = position;
+    int markerIndex = _markers.size();
+    beginInsertRows( QModelIndex(), markerIndex, markerIndex );
+    _markers.push_back( Marker( id, position ));
+    endInsertRows();
     emit( updated( shared_from_this( )));
 }
 
 void Markers::updateMarker( const int id, const QPointF position )
 {
-    if( !_markers.count( id ))
+    auto it = std::find_if( _markers.begin(), _markers.end(), [&id]( const Marker& marker )
+              { return marker.first == id; });
+
+    if( it  == _markers.end())
         return;
 
-    _markers[id] = position;
+    it->second = position;
     emit( updated( shared_from_this( )));
+
+    int markerIndex = it - _markers.begin();
+    emit dataChanged( createIndex( markerIndex, 0 ), createIndex( markerIndex, 0 ));
 }
 
 void Markers::removeMarker( const int id )
 {
-    if( !_markers.count( id ))
+    auto it = std::find_if( _markers.begin(), _markers.end(), [&id]( const Marker& marker )
```

Comment:
No, the find condition is always the same. Only one time you compare the result directly in the if(), which is less readable. It would be more readable with an:
auto it = _findMarker( id ) everywhere